### PR TITLE
Read builds from GCS and convert it into a Grid message

### PR DESCRIPTION
**Pull request by fejta**
/assign @rmmh @michelle192837

### CODE REVIEW

**Comment by fejta** [comment]
Paging @rmmh and @michelle192837 

/meow space

**Comment by michelle192837** [comment]
Not enough cats.

In all seriousness, looks fine to me but I'll let @rmmh stamp it for Go.

File: testgrid/cmd/updater/main.go [hunk 1]
```diff
@@ -18,14 +18,343 @@ package main
...
 )
 
+type Build struct {
+	Bucket  *storage.BucketHandle
+	Context context.Context
+	Prefix  string
+}
+
+type Started struct {
+	Timestamp   int64             `json:"timestamp"` // python time.time()
```
Comment (rmmh) [inline testgrid/cmd/updater/main.go]: maybe "epoch seconds" instead?

File: testgrid/cmd/updater/main.go [hunk 2]
```diff
@@ -18,14 +18,343 @@ package main
...
+		}
+		if r := ar.Remain(); r > 50e6 {
+			return nil, fmt.Errorf("too large: %s is %d > 50M", ap, r)
+		}
+		buf, err := ioutil.ReadAll(ar)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read all of %s: %v", ap, err)
+		}
+
+		var suites JunitSuites
```
Comment (rmmh) [inline testgrid/cmd/updater/main.go]: Can you extract this XML parsing (buf -> suite data) and add some unit tests?

Look a gubernator/view_build_test.py for some examples (`test_testsuites` and `test_nested_testsuites`).

File: testgrid/cmd/updater/main.go (hunk 1 above)
Comment (fejta) [reply testgrid/cmd/updater/main.go]: Done

File: testgrid/cmd/updater/main.go (hunk 2 above)
Comment (fejta) [reply testgrid/cmd/updater/main.go]: Done

Please note that this is not expected to create correct grid messages yet (that is coming in later PRs), and a lot of this is gets refactored (note that we throw away the failure message right now, for example)....

**Pull request merged**